### PR TITLE
refactor: `BinaryToOctal`

### PR DESCRIPTION
**Pull request by alxkm**
Improving the code to make it more readable.
Adding additional tests to this algorithm.

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/conversions/BinaryToOctal.java [hunk 1]
```diff
@@ -1,27 +1,16 @@
...
-
 /**
  * Converts any Binary number to an Octal Number
  *
  * @author Zachary Jones
  */
 public final class BinaryToOctal {
-    private BinaryToOctal() {
-    }
+    private static final int BITS_IN_GROUP = 3;
```
Comment (BamaCharanChhandogi) [inline src/main/java/com/thealgorithms/conversions/BinaryToOctal.java:9]: The constant `BITS_IN_GROUP ` could be renamed to `BITS_PER_OCTAL_DIGIT` to make its purpose more explicit and improve readability.

File: src/main/java/com/thealgorithms/conversions/BinaryToOctal.java [hunk 2]
```diff
@@ -31,20 +20,26 @@ public static void main(String[] args) {
      * @return The octal number
      */
     public static String convertBinaryToOctal(int binary) {
-        String octal = "";
-        int currBit = 0;
-        int j = 1;
+        if (binary == 0) {
+            return "0";
+        }
+
```
Comment (BamaCharanChhandogi) [inline src/main/java/com/thealgorithms/conversions/BinaryToOctal.java:26]: Consider adding input validation in the convertBinaryToOctal method to ensure that the input is a valid binary number (composed only of 0 and 1). If the input is invalid, the method could throw an IllegalArgumentException or handle it gracefully by returning an appropriate error message.

`if (!String.valueOf(binary).matches("[01]+")) {
    throw new IllegalArgumentException("Input is not a valid binary number.");
}
`

**Review by BamaCharanChhandogi — COMMENTED** [review]
Overall, the updated code looks great, but I suggest adding input validation to ensure the input is a valid binary number. Also, consider enhancing the test cases to cover potential edge cases like invalid inputs.

File: src/main/java/com/thealgorithms/conversions/BinaryToOctal.java (hunk 2 above)
Comment (alxkm) [reply src/main/java/com/thealgorithms/conversions/BinaryToOctal.java:22]: Good observation. Thanks!

File: src/main/java/com/thealgorithms/conversions/BinaryToOctal.java (hunk 1 above)
Comment (alxkm) [reply src/main/java/com/thealgorithms/conversions/BinaryToOctal.java]: Yes, this is more readable, thank you.

**Review by BamaCharanChhandogi — APPROVED** [review]
LGTM, Thanks.
